indico/queries/custom_blueprint.py

```diff
@@ -0,0 +1,98 @@
+import typing as t
+
+from indico import GraphQLRequest
+from indico.errors import IndicoInputError
+from indico.types.workflow import ComponentFamily
+from indico.types.custom_blueprint import TaskBlueprint
+
+
+class RegisterCustomBlueprint(GraphQLRequest):
+    query = """
+mutation createCustomBP(
+  $name: String!,
+  $description: String!,
+  $footer: String!,
+  $tags: [BlueprintTag]!,
+  $config: {config_type}!,
+  $icon: String,
+  $allAccess: Boolean,
+  $datasetIds: [Int]
+) {{
+  {mutation_name}(
+    name: $name,
+    description: $description,
+    footer: $footer,
+    tags: $tags,
+    config: $config,
+    icon: $icon,
+    allAccess: $allAccess,
+    datasetIds: $datasetIds
+  ) {{
+    id
+    name
+    componentFamily
+    icon
+    description
+    footer
+    tags
+    enabled
+    config
+  }}
+}}
+"""
+
+    def __init__(
+        self,
+        component_family: ComponentFamily,
+        name: str,
+        description: str,
+        config: t.Dict,
+        tags: t.List[str],
+        footer: str = "",
+        icon: str = None,
+        all_access: bool = None,
+        dataset_ids: t.List[int] = None,
+    ):
+        if not component_family or not name or not description or not config or not tags:
+            none_vals = []
+            if not component_family:
+                none_vals.append("component_family")
+            if not name:
+                none_vals.append("name")
+            if not description:
+                none_vals.append("description")
+            if not config:
+                none_vals.append("config")
+            if not tags:
+                none_vals.append("tags")
+            raise IndicoInputError(f"The following arguments cannot be None: {','.join(none_vals)}")
+        if not all([isinstance(t, str) for t in tags]):
+            raise IndicoInputError("'tags' must be a list of strings")
+        if not all(config.values()):
+            raise IndicoInputError("values in the 'config' dict must not be None")
+        if not isinstance(component_family, ComponentFamily):
+            raise IndicoInputError("'component_family' must be of type indico.types.workflows.ComponentFamily")
+        if component_family.name not in ["OUTPUT", "FILTER", "MODEL"]:
```

Comment:
this is the only one that we might want to leave out, in case the BE adds a new enum (maybe just check it's not none)?

Reply:
That makes sense, especially #2! For ease of code maintenance, is it possible to keep a supported component family list in the file with the enum?

Reply:
good idea!